sycl/source/detail/graph_impl.hpp

```diff
@@ -336,6 +353,9 @@ class node_impl {
 /// Implementation details of command_graph<modifiable>.
 class graph_impl {
 public:
+  /// Protects all the fields that can be changed by class' methods.
+  mutable std::recursive_mutex MMutexGraph;
```

Comment:
```suggestion
  mutable std::recursive_mutex MMutex;
```

I don't think `Graph` is necessary in the name here, since it is the sole mutex member of the graph. Same for the other mutexes.